src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -236,6 +241,20 @@ private void ValidateProcessedBlock(Block suggestedBlock, ProcessingOptions opti
         ReceiptsTracer.StartNewBlockTrace(block);
 
         _beaconBlockRootHandler.ApplyContractStateChanges(block, spec, _stateProvider);
+
+        if (spec.IsEip2935Enabled)
+        {
+            // TODO: find a better way to handle this - no need to have this check everytime
+            //      this would just be true on the fork block
+            BlockHeader parentHeader = _blockTree.FindParentHeader(block.Header, BlockTreeLookupOptions.None);
+            if (parentHeader is not null && parentHeader!.Timestamp < spec.Eip2935TransitionTimestamp)
```

Comment:
Other way of deciding would be just reading 1 slot from the store and depending if data was there decide do we need to add 1 item or init data, not sure which one is better.

Reply:
i am keeping the current impl for now